ext/hash/hash_sha_ni.c

```diff
@@ -27,11 +27,11 @@
 #include "php_hash.h"
 #include "php_hash_sha.h"
 
-#if (defined(__i386__) || defined(__x86_64__)) && defined(HAVE_IMMINTRIN_H)
+#if defined(PHP_HASH_INTRIN_SHA_NATIVE) || defined(PHP_HASH_INTRIN_SHA_RESOLVER)
 
 # include <immintrin.h>
 
-# if PHP_HASH_INTRIN_SHA_RESOLVER
+# if defined(HAVE_FUNC_ATTRIBUTE_TARGET)
```

Comment:
Is this still wrong? Shouldn't that be
```suggestion
# if  defined(PHP_HASH_INTRIN_SHA_RESOLVER) && defined(HAVE_FUNC_ATTRIBUTE_TARGET)
```

Reply:
It probably doesn't matter here [1], but I believe it should check both.

My understanding is that the `target()` attribute overrides any target flags provided by the user. If a user passes `-march=native`, supporting additional instruction sets, then the compiler optimizations would not take them into account for the PHP_HASH_INTRIN_SHA_NATIVE case.

[1] The function's contents are basically only calls to intrinsics, where there is no compiler optimization potential anyways.

Reply:
@cmb69 This thread is still unresolved.

Reply:
Oh, indeed. See also

https://github.com/php/php-src/blob/217ea732fc29fc478a30070be03b6d0222227550/Zend/zend_portability.h#L568-L570